ui/splash/src/commonMain/kotlin/social/androiddev/splash/navigation/RootComponent.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * This file is part of MastodonX.
+ *
+ * MastodonX is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * MastodonX is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with MastodonX. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.splash.navigation
+
+import com.arkivanov.decompose.ComponentContext
+import com.arkivanov.decompose.router.stack.ChildStack
+import com.arkivanov.decompose.router.stack.StackNavigation
+import com.arkivanov.decompose.router.stack.childStack
+import com.arkivanov.decompose.router.stack.push
+import com.arkivanov.decompose.router.stack.replaceCurrent
+import com.arkivanov.decompose.value.Value
+import com.arkivanov.essenty.parcelable.Parcelable
+import com.arkivanov.essenty.parcelable.Parcelize
+import social.androiddev.welcome.navigation.WelcomeScreenComponent
+
+/**
+ * The base component describing all business logic needed for the root entry point
+ */
+interface RootComponent {
+
+    // Store a stack of components and their configurations in this root graph
+    val childStack: Value<ChildStack<*, Child>>
+
+    /**
+     * Supported "Child"s in this navigation stack. These are created from a configuration that
+     * contains any arguments for this particular child in the navigation stack.
+     */
+    sealed class Child {
+        data class SplashScreenChild(val component: SplashComponent) : Child()
+
+        data class WelcomeScreenChild(val component: WelcomeScreenComponent) : Child()
+
+        object TimelineScreenChild : Child()
+
+        object SelectServerScreenChild : Child()
+    }
+
+    /**
+     * Supported deep links for this root graph
+     */
+    sealed interface DeepLink {
+        object None : DeepLink
+    }
+
+    companion object {
+
+        fun createDefaultComponent(
+            componentContext: ComponentContext,
+            deepLink: DeepLink = DeepLink.None,
+        ): RootComponent = DefaultRootComponent(
```

Comment:
Not sure if this is a preferred way in the project, but maybe decouple the interface from the implementation and just expose `DefaultRootComponent`. You will likely also have another implementations for Compose previews and/or integration tests.